plugins/pulp_rpm/plugins/importers/yum/sync.py

```diff
@@ -456,30 +456,33 @@ def update_content(self, metadata_files, url):
         :param url: curret URL we should sync
         :type: str
         """
-        rpms_to_download, drpms_to_download = self._decide_what_to_download(metadata_files)
+        catalog = PackageCatalog(self.conduit.importer_object_id, url)
```

Comment:
What do you think about making this an instance attribute, so it could be accessed anywhere as ``self.catalog`` ? That would remove the need to pass it around all over the place, and would be similar to how the conduit and other structures are accessed.